src/renderer/components/DirectoryPicker.tsx

```diff
@@ -57,57 +68,77 @@ export default class DirectoryPicker extends React.Component {
-    this.setState({rootDir: e.currentTarget.value});
+  toggleRemoveAllModal() {
+    this.setState({
+      removeAllIsOpen: !this.state.removeAllIsOpen
+    });
   };
 
-  toggleModal() {
+  toggleImportModal() {
     this.setState({
-      isOpen: !this.state.isOpen,
+      importIsOpen: !this.state.importIsOpen,
       importURL: ""
     });
   };
 
   doImport() {
     let importURL = this.state.importURL;
-    this.setState({
-      isOpen: !this.state.isOpen,
-      importURL: ""
-    });
+    let hastebinURL = this.state.importURL;
+    this.toggleImportModal();
     if (!importURL) {
       return;
     }
-    let rootDir = this.state.rootDir;
+    let rootDir = this.state.rootDir.toString();
     if (!rootDir.endsWith(sep)) {
       rootDir += sep;
     }
-    // Remove everything before "sources="
-    importURL = importURL.substring(importURL.indexOf("sources=") + 8);
-    // Remove everything after the sources parameter
-    importURL = importURL.substring(0, importURL.indexOf("&"));
-    // Split into blog names
-    let importURLs = importURL.split("%20");
-    // Append root onto each blog
-    for (let u = 0; u<importURLs.length; u++) {
-      let fullPath = rootDir + importURLs[u];
-      if (this.props.directories.includes(fullPath)) {
-        // Remove index and push u back
-        importURLs.splice(u,1);
-        u-=1
-      } else {
-        importURLs[u] = fullPath;
+    if (importURL.includes("sources=")) {
+      // Remove everything before "sources="
+      importURL = importURL.substring(importURL.indexOf("sources=") + 8);
+
+      if (importURL.includes("&")) {
+        // Remove everything after the sources parameter
+        importURL = importURL.substring(0, importURL.indexOf("&"));
+      }
+
+      // Split into blog names
+      let importURLs = importURL.split("%20");
+      // Append root onto each blog
+      for (let u = 0; u < importURLs.length; u++) {
+        let fullPath = rootDir + importURLs[u];
+        if (this.props.directories.includes(fullPath) || importURLs[u] === sep || importURLs[u] === "") {
+          // Remove index and push u back
+          importURLs.splice(u, 1);
+          u -= 1
+        } else {
+          importURLs[u] = fullPath;
+        }
       }
-    }
 
-    // Don't add empty blog
-    if (importURLs.length == 1 && importURLs[0] === sep) {
-      return;
-    } else {
       // Add list
       this.props.onChange(this.props.directories.concat(importURLs));
     }
+
+    if (hastebinURL.includes("pastebinId=")) {
+      // Remove everything before "sources="
+      hastebinURL = hastebinURL.substring(hastebinURL.indexOf("pastebinId=") + 11);
+
+      if (hastebinURL.includes("&")) {
+        // Remove everything after the sources parameter
+        hastebinURL = hastebinURL.substring(0, hastebinURL.indexOf("&"));
+      }
+
+      // Update hastebin URL (if present)
+      this.props.onChangeHastebinID(hastebinURL);
+    }
```

Comment:
In the future, it would be nice to decouple the directory picker and the URL importer, but it's fine for now. I'm pretty much trusting you on the URL code.

Reply:
Like just in it's own Component?

Reply:
Yeah, or pull the import modal out into `SceneDetail`. It just doesn't seem to belong here. The directory picker could be used to pick directories for any purpose, but the URL importer makes it a lot more specific.

Reply:
Done 👍